src/internals/legend.js

```diff
@@ -552,15 +586,56 @@ extend(ChartInternal.prototype, {
 			.attr("x", xForLegendRect)
 			.attr("y", yForLegendRect);
 
-		const tiles = $$.legend.selectAll(`line.${CLASS.legendItemTile}`)
-			.data(targetIdz);
 
-		(withTransition ? tiles.transition() : tiles)
-			.style("stroke", $$.color)
-			.attr("x1", x1ForLegendTile)
-			.attr("y1", yForLegendTile)
-			.attr("x2", x2ForLegendTile)
-			.attr("y2", yForLegendTile);
+		if (!hasCustomPoints) {
+			const tiles = $$.legend.selectAll(`line.${CLASS.legendItemTile}`)
+				.data(targetIdz);
+
+
+			(withTransition ? tiles.transition() : tiles)
+				.style("stroke", $$.color)
+				.attr("x1", x1ForLegendTile)
+				.attr("y1", yForLegendTile)
+				.attr("x2", x2ForLegendTile)
+				.attr("y2", yForLegendTile);
+		} else {
+			const tiles = $$.legend.selectAll(`.${CLASS.legendItemPoint}`)
+				.data(targetIdz);
+
+			(withTransition ? tiles.transition() : tiles)
+				.each((data, idx, selection) => {
+					const node = selection[idx];
+					const nodeName = node.nodeName.toLowerCase();
+					const d3Node = d3Select(node);
+					let x = "x";
+					let y = "y";
+					let yOffset = 2;
```

Comment:
The size for circle and rectangle point displayed in the legend, is quite small.
How about increase little bit more?

```js
let yOffset = 2.5;
let radius;
let width;
let height;

if (nodeName === "circle") {
	x = "cx";
	y = "cy";
	radius = $$.config.point_r + ($$.config.point_r * 0.2);
	yOffset = -($$.config.point_r);
} else if (nodeName === "rect") {
	width = $$.config.point_r * 2.5;
	height = $$.config.point_r * 2.5;
}
```